app/models/pawn.rb

```diff
@@ -1,6 +1,29 @@
 class Pawn < Piece
-  def move_valid?
-    # add logic here
+  def move_valid?(x, y)
+    return false if space_occupied?(x, y) == true
+    return false if legal_move?(x, y) == false
```

Comment:
Cool, although you can still just add those rules to `move_valid?`, as in both `legal_move?` and `legal_move?` are doing roughly the same thing, so most likely all the contents of `legal_move?` should just become part of `move_valid?`.